_posts/2020-12-24-how-to-start-with-foreman

```diff
@@ -0,0 +1,399 @@
+run fails. You can also manage the configuration management policy, be it Ansible, Puppet,
+Salt or Chef. With [Katello](https://theforeman.org/plugins/katello/3.18/index.html) plugin,
+you can watch what packages are upgradable or what security vulnerabilities were detected and
+can be patched on your hosts. You can even manage what package versions or files your hosts
+have access to. You can define OpenSCAP based compliance policy and start enforcing it across
+your infrastructure.
+
+All these use-cases deserve it's own post, but we'll start first with how to
+begin, or better to say, how to create some hosts.
+
+### Adding unmanaged hosts manually
+
+The easiest way to create an unmanaged host is using [Hammer CLI](https://www.theforeman.org/manuals/latest/index.html#4.5CommandLineInterface).
+Once Hammer is installed and configured, you can create a host with this command
+
+```
+hammer host create --name my-first-host \
+  --managed false \
+  --build false \
+  --organization 'Default Organization' \
+  --location 'Default Location'
+```
+
+Name is usually the FQDN of the host, but it can be also just a hostname without domain.
+The `managed` parameter tells Foreman, this will be unmanaged host. The build value
+represents whether the host is currently in build mode, which is only possible
+for managed hosts. Without explaining what exactly it means, it always needs to be set
+to false for unmanaged hosts, otherwise the host record is considered invalid.
+
+More parameters, such as `--ip`, `--owner`, `--comment`, `--model` can be specified.
+For the full list, run `hammer host create --help`
+
+The other option is to use our API directly. In fact, all hammer commands are using
+REST API that Foreman provides. For more information, see [our manual](https://www.theforeman.org/manuals/latest/index.html#5.1API).
+The equivalent of the hammer command would be the following request
+
+```
+curl -X POST https://foreman.example.tst/api/hosts \
+  -d '{"name": "my-first-host", "managed": false, "organization_id": "2", "location_id": "1"}' \
+  -u admin:changeme -H "Content-Type: application/json"
+```
+
+Currently, there's no way how to manually create unmanaged host easily in
+Foreman Web UI. You can create only the managed host and turn it later to unmanaged.
+Managed hosts require much more information from user, so that Foreman knows,
+how to configure external services. Therefore it's good to start with unmanaged
+hosts first, unless you plan to provision hosts as the first step.
+
+Once you create the unmanaged host, you can work with it in the UI. Navigate to
+*Hosts -> All Hosts* page and see the entry with name my-first-host. You can click
+Edit button and fill in more information about the host. Clicking on it's name
+takes you to its detail page.
+
+<p float='left' align='center'>
+  <img src="/static/images/blog_images/2020-12-24-how-to-start-with-foreman/hosts.png" width="1000" />
```

Comment:
1000px? that sounds a lot.

Reply:
yeah, that's the only readable on my monitor, however I can shrink it to 700 which I've seen elsewhere